cleanlab/experimental/datalab/issue_manager/duplicate.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union, cast
+
+import numpy as np
+import pandas as pd
+from sklearn.neighbors import NearestNeighbors
+from sklearn.utils.validation import check_is_fitted
+
+from cleanlab.experimental.datalab.issue_manager import IssueManager
+
+if TYPE_CHECKING:  # pragma: no cover
+    from cleanlab import Datalab
+
+
+class NearDuplicateIssueManager(IssueManager):
+    """Manages issues realted to near-duplicate examples."""
+
+    issue_name: str = "near_duplicate"
+
+    def __init__(
+        self,
+        datalab: Datalab,
+        metric: Optional[str] = "cosine",
```

Comment:
I heard talk of defaulting to a Euclidean metric when the feature array is fewer than 3 dimensions. Should we implement that in `NearDuplicateIssueManager` and `OutOfDistributionIssueManager` as well? If we do, then it should be implemented for any issue manager that relies on the KNN graph.

Reply:
I think it's fine to add this to `NearDuplicateIssueManager`.
`OutOfDistributionIssueManager` behaves this way currently, but the selection of the (default) distance metric is now "abstracted" away with `OutOfDistribution`.
https://github.com/cleanlab/cleanlab/blob/66e0df088f40ec4f2ee9dfed01c370f76ea38c6b/cleanlab/experimental/datalab/issue_manager/outlier.py#L29

It will take more effort to unify this behavior for all managers that rely on the KNN graph. We can look at it at a later time.

Reply:
great catch @jecummin !

Ya @elisno let's just add it manually also in NearDuplicateIssueManager and revisit reducing duplicate code later :P